system/CodeIgniter.php

```diff
@@ -432,7 +432,10 @@ protected function handleRequest(?RouteCollectionInterface $routes, Cache $cache
             return $returnResponse ? $possibleResponse : $possibleResponse->pretend($this->useSafeOutput)->send();
         }
 
-        if ($possibleResponse instanceof Request) {
+        if (
+            $possibleResponse instanceof IncomingRequest
```

Comment:
While I think it unlikely, what if someone had created their own Request class? LIke perhaps a more PSR-friendly version. What is the benefit of this change other than limiting those options for devs?